cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll

```diff
@@ -100,3 +100,23 @@ class StdSequenceContainerSwap extends TaintFunction {
     output.isQualifierObject()
   }
 }
+
+/**
+ * The standard container functions `at` and `operator[]`.
+ */
+class StdSequenceContainerAt extends TaintFunction {
+  StdSequenceContainerAt() {
+    this.hasQualifiedName("std", "vector", "at") or
+    this.hasQualifiedName("std", "vector", "operator[]") or
+    this.hasQualifiedName("std", "array", "at") or
+    this.hasQualifiedName("std", "array", "operator[]") or
+    this.hasQualifiedName("std", "deque", "at") or
+    this.hasQualifiedName("std", "deque", "operator[]")
```

Comment:
We can now write this as

```ql
this.hasQualifiedName("std", ["vector", "array", "deque"], ["at", "operator[]"])
```

It's a matter of personal preference which one is best -- they should be completely equivalent.

Reply:
In similar cases we are not always looking for the cartesian product, for example:
```
    this.hasQualifiedName("std", "vector", "push_back") or
    this.hasQualifiedName("std", "deque", "push_back") or
    this.hasQualifiedName("std", "deque", "push_front") or
    this.hasQualifiedName("std", "list", "push_back") or
    this.hasQualifiedName("std", "list", "push_front") or
    this.hasQualifiedName("std", "forward_list", "push_front")
```
But actually that reads quite well as:
```
    this.hasQualifiedName("std", "vector", "push_back") or
    this.hasQualifiedName("std", "deque", ["push_back", "push_front"]) or
    this.hasQualifiedName("std", "list", ["push_back", "push_front"]) or
    this.hasQualifiedName("std", "forward_list", "push_front")
```
I'll make the changes shortly.

Reply:
Done.